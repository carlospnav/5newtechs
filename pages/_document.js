import Document, { Head, Main, NextScript } from "next/document"
import { ServerStyleSheet } from "styled-components"

export default class MyDocument extends Document {
  static async getInitialProps({ renderPage }) {
    const page = renderPage()
    return { ...page }
  }

  render() {
    const sheet = new ServerStyleSheet()
    const main = sheet.collectStyles(<Main />)
    const styleTags = sheet.getStyleElement()

    return (
      <html lang="en">
        <Head>
          <meta charSet="utf-8" />
          <meta
            name="viewport"
            content="width=device-width, initial-scale=1.0, minimum-scale=1.0, maximum-scale=1.0, user-scalable=no"
          />
          <title>Star Wars People</title>
          <link href="https://fonts.googleapis.com/css?family=Roboto:100,300,500,900" rel="stylesheet" />
          {styleTags}
        </Head>
        <body>
          {main}
          <NextScript />
        </body>
      </html>
    )
  }
}
