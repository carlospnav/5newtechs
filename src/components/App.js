import styled from "styled-components"
import "theme/global"

const Main = styled.div`
  display: flex;
  flex-direction: column;
`

export default ({ children }) => <Main>{children}</Main>
