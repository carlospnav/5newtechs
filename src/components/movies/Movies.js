import { graphql } from "react-apollo"
import { compose, branch, withProps, renderComponent } from "recompose"
import withUrlParam from "lib/withUrlParam"
import { fetchPolicyList } from "lib/constants"
import movieQuery from "queries/movie"
import Spinner from "../common/Spinner"

export function Movies(props) {
  console.log(props);
  return (
    <p>
      Movies! oi!
    </p>
  )
}

export default compose(
  withUrlParam("filmId"),
  graphql(movieQuery, { options: fetchPolicyList }),
  branch(({data}) => data.loading, renderComponent(Spinner)),
  withProps(({ data: { film } }) => ({
    film: film
  }))
)(Movies)