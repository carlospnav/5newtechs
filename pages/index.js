import { compose } from "recompose"
import withData from "lib/withData"
import withSetUrlParams from "lib/withSetUrlParams"
import PeopleList from "components/PeopleList"
import App from "components/App"

function IndexPage({ url: { pathname } }) {
  return (
    <App>
      <PeopleList />
    </App>
  )
}

export default compose(withData, withSetUrlParams)(IndexPage)
