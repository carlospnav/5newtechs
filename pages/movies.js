import { compose } from "recompose"
import App from "components/App"
import Movie from "components/movie/Movie"
import withData from "lib/withData"
import withSetUrlParams from "lib/withSetUrlParams"

function MoviesPage({ url: { pathname } }){
  return (
    <App>
      <Movie />
    </App>
  )
}

export default compose(withData, withSetUrlParams)(MoviesPage)