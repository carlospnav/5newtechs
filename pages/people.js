import { compose } from "recompose"
import withData from "lib/withData"
import App from "components/App"
import PeopleListContainer from "components/people/PeopleListContainer"
import withSetUrlParams from "lib/withSetUrlParams"
import MainContainer from "../src/components/common/MainContainer";
import Background from "../src/components/common/background/Background";
import PageLogo from "../src/components/common/pageLogo/PageLogo";
import Header from '../src/components/common/header/Header';

function PeoplePage({ url: { pathname } }) {
  return (
    <App>
      <Background />
      <Header main>
        <PageLogo />
      </Header>
      <MainContainer>
        <PeopleListContainer />
      </MainContainer>
    </App>
  )
}

export default compose(withData, withSetUrlParams)(PeoplePage)
