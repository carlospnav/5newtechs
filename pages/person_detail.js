import { compose } from "recompose"
import withData from "lib/withData"
import withSetUrlParams from "lib/withSetUrlParams"
import App from "components/App"
import MainContainer from "common/mainContainer/MainContainer"
import Background from "common/background/Background"
import Header from "common/header/Header"
import PageLogo from "common/pageLogo/PageLogo"
import PersonDetail from "components/details/PersonDetail"

export function PersonDetailPage({ url: { pathname } }) {
  return (
    <App>
      <Background small/>
      <Header>
        <PageLogo small/>
      </Header>
      <MainContainer details>
        <PersonDetail />
      </MainContainer>
    </App>
  )
}

export default compose(withData, withSetUrlParams)(PersonDetailPage)
