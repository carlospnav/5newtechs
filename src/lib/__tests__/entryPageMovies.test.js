import { shallow } from "enzyme"
import { MoviesPage } from "../../../pages/movies"
import App from "components/App"

describe("<movies>", () => {
  let wrapper;

  function movies(){
    wrapper = shallow(<MoviesPage url={{pathname: "movies"}} />)
  }

  beforeEach(movies)
  afterEach(() => wrapper = undefined)

  it("should render the proper component tree as expected.", () => {
    expect(wrapper.find(App)).toMatchSnapshot()
  })
})