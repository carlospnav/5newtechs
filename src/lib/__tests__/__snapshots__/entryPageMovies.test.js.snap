// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<movies> should render the proper component tree as expected. 1`] = `
<Component>
  <Connect(Apollo(branch(withProps(Movies)))) />
</Component>
`;
